internal/jsnum/jsnum_test.go

```diff
@@ -286,12 +316,27 @@ func TestRemainder(t *testing.T) {
 		{123, negativeZero, NaN()},
 		{0, 123, 0},
 		{negativeZero, 123, negativeZero},
+		{-1, 2, -1},
+		// more exhaustive tests for zero and negative zero
+		{negativeZero, 1, negativeZero},
+		{negativeZero, 2, negativeZero},
+		{1, 1, 0},
+		{1, 2, 1},
+		{-1, 1, negativeZero},
+		{5, 2, 1},
+		{5, -2, 1},  // sign of divisor shouldn't matter
+		{-5, 2, -1}, // retain sign of dividend
+		{-5, -2, -1},
+		// edge cases with small numbers
+		{1e-10, 1, 1e-10},
+		{1, 1e-10, 0}, // should be 0, or very close to 0
 	}
 
 	for _, test := range tests {
 		t.Run(fmt.Sprintf("%v %% %v", test.x, test.y), func(t *testing.T) {
 			t.Parallel()
-			assertEqualNumber(t, test.x.Remainder(test.y), test.want)
+			actual := test.x.Remainder(test.y)
```

Comment:
This is an unnecessary refactor. I don't really get what the point of some of these changes is...